jerry-core/jcontext/jcontext.c

```diff
@@ -19,6 +19,50 @@
  * @{
  */
 
+#ifdef JERRY_ENABLE_EXTERNAL_CONTEXT
+
+/**
+ * Get the current context from the current instance.
+ *
+ * @return the pointer to the context
+ */
+jerry_context_t *
+jerry_context_get_current_context (void)
```

Comment:
"its performance affects the global performance a little." - I am not fully agree with this. People do things on JS level many times.